### DISCUSSION TITLE
Cloning Git repositories through session parameters

### DISCUSSION THREAD

**Discussion by robertjndw** [eclipse-theia/theia-cloud#299] [Q&A] [answered]
Hello everyone,
I have a question regarding the custom session parameters stated in the documentation (see [here](https://main--theia-cloud.netlify.app/documentation/moredocumentation/)). I want to start with an empty workspace and add a repository by its URL when opening the session. As far as I can see, this can be achieved using these session parameters. 
If I am not mistake, I need to read the respective variables in my Theia extension. My question now is, is there a simpler way to achieve this, other than writing my own Theia extension? Is there an existing Git extension which I can utilize for cloning through parameters?

**Comment by lucas-koehler** [comment, marked as answer]
Hello @robertjndw ,
Yes you can hand in the URL via the custom session parameters and then read them as environment variables in Theia.
Unfortunately, we are not aware of an existing Theia or VSCode extension that does this for you. Thus, you need to implement a Theia extension to do this.
In your extension implementation, you can conveniently access environment variables via the `EnvVariablesServer`.

E.g. you could inject it as a field in a class like this:
```ts
@inject(EnvVariablesServer)
protected environment: EnvVariablesServer;
```
